src/transformers/models/owlv2/modeling_owlv2.py

```diff
@@ -1634,6 +1623,110 @@ def image_guided_detection(
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> query_url = "http://images.cocodataset.org/val2017/000000058111.jpg"
+        >>> query_image = Image.open(requests.get(query_url, stream=True).raw)
+        >>> inputs = processor(images=image, query_images=query_image, return_tensors="pt")
+        >>> with torch.no_grad():
+        ...     outputs = model.image_guided_detection(**inputs)
+        >>> # Target image sizes (height, width) to rescale box predictions [batch_size, 2]
+        >>> target_sizes = torch.Tensor([image.size[::-1]])
+        >>> # Convert outputs (bounding boxes and class logits) to COCO API
+        >>> results = processor.post_process_image_guided_detection(
+        ...     outputs=outputs, threshold=0.98, nms_threshold=0.95, target_sizes=target_sizes
+        ... )
+        >>> i = 0  # Retrieve predictions for the first image
+        >>> boxes, scores = results[i]["boxes"], results[i]["scores"]
+        >>> for box, score in zip(boxes, scores):
+        ...     box = [round(i, 2) for i in box.tolist()]
+        ...     print(f"Detected similar object with confidence {round(score.item(), 3)} at location {box}")
+        Detected similar object with confidence 1.0 at location [341.67, 17.54, 642.32, 278.51]
+        Detected similar object with confidence 1.0 at location [6.75, 38.97, 326.62, 354.85]
+        ```"""
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+
+        # Compute feature maps for the input and query images
+        query_feature_map = self.image_embedder(pixel_values=query_pixel_values)[0]
+        feature_map, vision_outputs = self.image_embedder(
+            pixel_values=pixel_values,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+        )
+
+        batch_size, num_patches, num_patches, hidden_dim = feature_map.shape
+        image_feats = torch.reshape(feature_map, (batch_size, num_patches * num_patches, hidden_dim))
+
+        batch_size, num_patches, num_patches, hidden_dim = query_feature_map.shape
+        query_image_feats = torch.reshape(query_feature_map, (batch_size, num_patches * num_patches, hidden_dim))
+
+        # Get boxes and class embeddings of query image
+        query_pred_boxes = self.box_predictor(query_image_feats, feature_map=query_feature_map)
+        query_class_embeddings = self.class_predictor(query_image_feats)[1]
+
+        # v2 differs from v1 in that we use the objectness head to predict the objectness of the patches of the query image
```

Comment:
Is this the naming that the model uses i.e. does the paper use "image_guided_detection_v1", "image_guided_detection_v2"? 

As so much of this logic is shared - couldn't we control this with a flag in the original `image_guided_detection` method?